autogen/agentchat/contrib/capabilities/transforms.py

```diff
@@ -53,13 +53,16 @@ class MessageHistoryLimiter:
     It trims the conversation history by removing older messages, retaining only the most recent messages.
     """
 
-    def __init__(self, max_messages: Optional[int] = None):
+    def __init__(self, max_messages: Optional[int] = None, keep_first_message: bool = False):
```

Comment:
Hello, as @marklysze pointed out `keep_first_message` seems necessary in order to ensure that the task is understood and the initial message not removed. Currently, the code alone does not provide clear guidance on when to set this parameter to `True`. Therefore, additional documentation on this aspect would be beneficial to clarify its usage.

Reply:
I'll mark this as resolved for now. We can add another PR to enhance if needed.